Add UICR.SECURESTORAGE configuration based on device tree partitions

SebastianBoe opened the pull request:
Add UICR.SECURESTORAGE configuration based on device tree partitions.
Validates partition layout and populates size fields in 1KB units.
Handles missing partitions gracefully.

Update the default memory map to include a `secure_storage_partition`,
which is divided into at most four subpartitions. These will be used to
configure UICR.SECURESTORAGE, if enabled.

SebastianBoe committed:
boards: nrf54h20dk: Add secure storage partitions

Update the default memory map to include a `secure_storage_partition`,
which is divided into at most four subpartitions. These will be used to
configure UICR.SECURESTORAGE, if enabled.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

jonathannilsen reviewed (commented):
Looks good.
I think we should address the part where we default to zeroes for partitions that we should expect to exist, apart from that the comments are just minor suggestions.

jonathannilsen commented on soc/nordic/common/uicr/gen_uicr.py:
```diff
@@ -198,6 +207,78 @@ class Uicr(c.LittleEndianStructure):
+        PartitionInfo(args.cpurad_its_address, args.cpurad_its_size, "cpurad_its_partition"),
+    ]
+
+    # Filter out zero-sized partitions (missing partitions)
+    present_partitions = [p for p in partitions if p.size > 0]
+
+    if not present_partitions:
+        # No partitions present - this is valid
+        return
+
+    # Check that the first present partition starts at the secure storage address
+    first_partition = present_partitions[0]
+    if first_partition.address != args.securestorage_address:
+        raise ScriptError(
+            f"First partition {first_partition.name} starts at {first_partition.address}, "
+            f"but must start at secure storage address {args.securestorage_address}"
+        )
```
The start address and combined size of the partitions must also be aligned to 4 kB, or else you will get a boot error. I think it is worth adding checks for that as well, so that we catch that at build time instead.

We could maybe require that the combined size of the subpartitions is equal to `secure_storage_partition`, to ensure that the definition is coherent?

I think this part can also be dropped, which might make it easier to implement those other checks:
```python
if not present_partitions:
    # No partitions present - this is valid
    return
```
Unless I'm missing something, if you have set up the devicetree in that way (i.e. defined `secure_storage_partition` without any subpartitions) then you have probably made some sort of error.

SebastianBoe committed:
soc: nordic: uicr: Add support for SECURESTORAGE

Add UICR.SECURESTORAGE configuration based on device tree partitions.
Validates partition layout and populates size fields in 1KB units.
Handles missing partitions gracefully.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>
